src/subcommand/wallet/transaction_builder.rs

```diff
@@ -1070,4 +1102,31 @@ mod tests {
       })
     )
   }
+
+  #[test]
+  fn send_inscriptions_with_custom_fee_rate() {
+    let utxos = vec![(outpoint(1), Amount::from_sat(10_000))];
+
+    pretty_assert_eq!(
+      TransactionBuilder::build_transaction(
+        satpoint(1, 0),
+        BTreeMap::from([(
+          satpoint(1, 0),
+          "bed200b55adcf20e359bbb762392d5106cafbafc48e55f77c94d3041de3521da"
+            .parse()
+            .unwrap()
+        )]),
+        utxos.into_iter().collect(),
+        recipient(),
+        vec![change(0), change(1)],
+        25.8,
+      ),
+      Ok(Transaction {
+        version: 1,
+        lock_time: PackedLockTime::ZERO,
+        input: vec![tx_in(outpoint(1))],
+        output: vec![tx_out(4_324, recipient())],
```

Comment:
This does check the free rate, but let's also add an explicit check where we calculate the fee rate on the actual returned transaction.